### DISCUSSION TITLE
CreateCopyJobsAsync in Move mode writes to Recycle Bin?

### DISCUSSION THREAD

**Discussion by ConaxLiu** [pnp/pnpcore#983] [Q&A] [answered]
Hi, using SharePoint out of box Move functionality, i can move a folder from one library to another library fairly quickly. However, when I use ISite.CreateCopyJobAsync, the folder is moved much slower, and I noticed the moved folder and the files in the folder also got copied into the site Recycle Bin.

                var copyJob = await context.Site.CreateCopyJobsAsync(itemUris, destinationAbsoluteUrl, new CopyMigrationOptions
                {
                    AllowSchemaMismatch = true,
                    AllowSmallerVersionLimitOnDestination = true,
                    IgnoreVersionHistory = true,
                    IsMoveMode = true,
                    BypassSharedLock = true,
                    ExcludeChildren = false,
                    NameConflictBehavior = SPMigrationNameConflictBehavior.KeepBoth
                }, true);

I would expect the CreateCopyJobsAsync works the same as the out of box Move functionality, but it looks like it is actually doing a Copy from source to destination first, and then a Delete of the source? Am I missing something, or perhaps there's another way the PnP Core can move files and folders without writting them into the Recycle Bin?

Thank you.

**Comment by jansenbe** [comment, marked as answer]
@ConaxLiu : the site copy feature uses the `_api/site/CreateCopyJobs` SPO REST call, what you see is the behavior as it was intended by the SharePoint team that provided this API, in PnP Core SDK we're just wrapping it.  The reason why it's slower is because this is an asynchronous process, there's a background job in SharePoint that handles these copy/moves. If you want it different you can also copy/move files individually, see https://pnp.github.io/pnpcore/using-the-sdk/files-intro.html#copying-and-moving-files.

**Reply by ConaxLiu** [reply]
Thank you for the reply @jansenbe
